src/Analyzers/AnalyzerFormatter.cs

```diff
@@ -130,40 +132,121 @@ static void LogDiagnosticLocations(Solution solution, IEnumerable<Diagnostic> di
+
+            // We need to run each codefix iteratively so ensure that all diagnostics are found and fixed.
+            foreach (var diagnosticId in reportedDiagnostics)
             {
+                var analyzers = analyzersById[diagnosticId];
+                var codefixes = fixersById[diagnosticId];
+
+                // If there is no codefix, there is no reason to run analysis again.
+                if (codefixes.IsEmpty)
+                {
+                    logger.LogWarning(Resources.Unable_to_fix_0_No_associated_code_fix_found, diagnosticId);
+                    continue;
+                }
+
                 var result = new CodeAnalysisResult();
                 foreach (var project in solution.Projects)
                 {
-                    if (!projectDiagnostics.TryGetValue(project.Id, out var diagnosticIds) ||
-                        !analyzer.SupportedDiagnostics.Any(diagnostic => diagnosticIds.Contains(diagnostic.Id)))
+                    // Only run analysis on projects that had previously reported the diagnostic
+                    if (!projectDiagnostics.TryGetValue(project.Id, out var diagnosticIds))
                     {
                         continue;
                     }
 
-                    await _runner.RunCodeAnalysisAsync(result, analyzer, project, formattablePaths, logger, cancellationToken).ConfigureAwait(false);
+                    await _runner.RunCodeAnalysisAsync(result, analyzers, project, formattablePaths, severity, logger, cancellationToken).ConfigureAwait(false);
                 }
 
                 var hasDiagnostics = result.Diagnostics.Any(kvp => kvp.Value.Count > 0);
-                if (hasDiagnostics && codefix != null)
+                if (hasDiagnostics)
                 {
-                    solution = await _applier.ApplyCodeFixesAsync(solution, result, codefix, logger, cancellationToken).ConfigureAwait(false);
-                    var changedSolution = await _applier.ApplyCodeFixesAsync(solution, result, codefix, logger, cancellationToken).ConfigureAwait(false);
-                    if (changedSolution.GetChanges(solution).Any())
+                    foreach (var codefix in codefixes)
                     {
-                        solution = changedSolution;
+                        var changedSolution = await _applier.ApplyCodeFixesAsync(solution, result, codefix, logger, cancellationToken).ConfigureAwait(false);
+                        if (changedSolution.GetChanges(solution).Any())
+                        {
+                            solution = changedSolution;
+                            break;
+                        }
                     }
                 }
             }
 
             return solution;
+
+            static ImmutableDictionary<string, ImmutableArray<DiagnosticAnalyzer>> CreateAnalyzerMap(
+                ImmutableArray<string> diagnosticIds,
+                ImmutableArray<DiagnosticAnalyzer> analyzers)
```

Comment:
looks like we use an inconsistent wrapping style. Which one do we prefer?

Reply:
I like this one because it doesn't change when things are renamed/refactored.